pkg/webhook/resources/virtualmachine/validator.go

```diff
@@ -239,6 +242,22 @@ func (v *vmValidator) checkVMBackup(vm *kubevirtv1.VirtualMachine) error {
 	return nil
 }
 
+func (v *vmValidator) checkTerminationGracePeriodSeconds(vm *kubevirtv1.VirtualMachine) error {
+	TerminationGracePeriodSeconds := vm.Spec.Template.Spec.TerminationGracePeriodSeconds
+
+	//Even mutator prevents this from happening, we still check it for robustness
```

Comment:
Can you explain the comment more, why mutator prevents this happening. Thanks.

Reply:
From my understanding, if VM's `TerminationGracePeriodSeconds` is not specified, vmMutator's [`patchTerminationGracePeriodSeconds()`](https://github.com/harvester/harvester/blob/5a7cf30480868276009877186bf0326df05cf7c7/pkg/webhook/resources/virtualmachine/mutator.go#L383) function will set VM's `TerminationGracePeriodSeconds` as `default-vm-termination-grace-period-seconds`. I will add more details in this comment. Thanks

Reply:
Thanks for the explanation. The guard here makes sense, we should not assume there is always a mutation.

Reply:
I just found that the VM's `TerminationGracePeriodSeconds` still could be nil, if it's updated by config yaml directly without specifying the `TerminationGracePeriodSeconds` field. I will update the comment according to this, too. thanks.